### ISSUE TITLE
Neomake not working in neovim for windows, when using bash as shell

### ISSUE THREAD

**Issue by indiofish** [neomake/neomake#2317]
### Expected behavior
Neomake in Neovim for windows using bash as shell
should behave just like
Neomake in Neovim for windows using cmd as shell

### Steps to reproduce

1. set shell=bash for neovim in Windows10.
2. :Neomake a python file.

### Output from :NeomakeInfo
```
Neomake: [15.21:1:1] Starting async job [string]: bash /s /c python "C:\Users\foo\.config\nvim\plugged\neomake\autoload\neomake\makers\ft\python\compile.py" best.py.
Neomake: [15.21:1:1] cwd: C:\Users\foo\sandbox.
Neomake: [15.21:1:1] output on stderr: ['/usr/bin/bash: /s: No such file or directory', ''].
Neomake: [15.21:1:1] exit: python: 127.
Neomake: [15.21:1:1] python: completed with exit code 127.
Neomake: [15.21:1:1] python: unexpected output on stderr: /usr/bin/bash: /s: No such file or directory.
Neomake: python: unexpected output on stderr: /usr/bin/bash: /s: No such file or directory
Neomake: [15.21:1:1] python: unexpected output. See :messages for more information.
Neomake: [15.21:1:1] Cleaning jobinfo.
```

By the way things look, seems like Neomake is passing the same string to bash that is passed when using cmd as shell. (The /s /c part)

Thanks in advance.

**Comment by blueyed** [comment]
Does it work with `set shell=bash`?

I've started a while ago to get tests running on AppVeyor for Windows, but it is a mess: https://github.com/neomake/neomake/pull/1575

**Comment by blueyed** [comment]
Also related: https://github.com/neomake/neomake/pull/1896

**Comment by blueyed** [comment]
And https://github.com/neomake/neomake/issues/1893.

**Comment by indiofish** [comment]
Do you mean if it works with
set shell=cmd ?

If not, no. This bug was produced with set shell=bash.
If so, Yes. I am not having any issues with set shell=cmd

**Comment by blueyed** [comment]
> If so, Yes. I am not having any issues with set shell=cmd

Ah, at least.

I do not have much motivation myself to fix Windows issues, but will gladly accept a PR - please review the referenced issues/PRs though.. a fix might be in there already.

**Comment by indiofish** [comment]
Looks like this is a Neovim Issue;
The value 'shellcmdflag' should be set according to the value of 'shell', but apparently it isn't.
Closing the Issue.

**Comment by indiofish** [comment]
https://github.com/neomake/neomake/blob/53284f8750e095ee3c1c4506379a6536f796bbde/autoload/neomake/utils.vim#L66-L68

```vim
return &shell=='cmd' && (has('win32') || has('win64'))
```
Assuming that the user is not using windows when they are not using the 'cmd' shell seems to work(at least for me)
Should I make a PR?

**Comment by blueyed** [comment]
No, this gets used also in other places, e.g. to use "NUL" instead of "/dev/null".
After all, you are still running Windows.
